### PR TITLE
Splash Screen

### LINKED ISSUE #68: screen: SplashScreen
# SplashScreen
### UI:  
animating LoadingWidget (#80) with gradient background ([colors](https://drive.google.com/file/d/1YI45JJuQiF7sZFT_V-Rz66dlPlNPSJv2/view?usp=sharing))

### Logic: 
It should check if the user is logged in (using sharedpreferences) and redirect them to HomeScreen or LoginScreen correspondingly.

### PR DESCRIPTION
**Pull request by Hard-Coder05**
### Description

Splash Screen UI and Logic

**Type of Change:** (Delete irrelevant options)
- Code
- User Interface

**Code/Quality Assurance Only** (Delete irrelevant options)

- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?

Tested locally.

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] Any dependent changes have been merged
- [ ] I have added one/more required dependencies
- [x] I have formatted the code using dart format (dartfmt)

**Code/Quality Assurance**
- [x] My changes generate no new warnings 
- [ ] My PR currently breaks something (fix or feature that would cause existing functionality to not work as expected)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

https://user-images.githubusercontent.com/54059881/107914113-c4ebfc00-6f87-11eb-8edd-8d0e01a4175d.mp4

### CODE REVIEW

File: lib/screens/splash/splash.dart [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:shared_preferences/shared_preferences.dart';
+
+class SplashScreen extends StatefulWidget {
+  @override
+  _SplashScreenState createState() => _SplashScreenState();
+}
+
+class _SplashScreenState extends State<SplashScreen> {
+  final splashDelay = 3;
```
Comment (werainkhatri) [inline lib/screens/splash/splash.dart]: should be in `dimens.dart`

File: lib/screens/splash/splash.dart [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+  @override
+  void initState() {
+    super.initState();
+    _loadWidget();
+  }
+
+  _loadWidget() async {
+    var _duration = Duration(seconds: splashDelay);
+    return Timer(_duration, navigationPage);
+  }
```
Comment (werainkhatri) [inline lib/screens/splash/splash.dart]: 1. no need to create small functions and call them. place these lines in `initState` directly.
2. don't create unwanted variables
3. call the `init()` function of main.dart (line 9) here.
```suggestion
  _loadWidget() async {
    await init();
    Timer(Duration(seconds: splashDelay), navigationPage);
  }
```

File: lib/screens/splash/splash.dart [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+  void navigationPage() {
+    final SharedPreferences sharedPreferences = sl.get<SharedPreferences>();
+    String token = sharedPreferences.getString(S.tokenKey);
```
Comment (werainkhatri) [inline lib/screens/splash/splash.dart]:
```suggestion
    String token = sl.get<SharedPreferences>().getString(S.tokenKey);
```

File: lib/screens/splash/splash.dart [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+  Widget build(BuildContext context) {
+    return Scaffold(
+        body: Stack(
+      children: [
+        ScreenBackground(elementId: 0),
+        Center(
+          child: ECellLogoAnimation(),
+        )
+      ],
+    ));
```
Comment (werainkhatri) [inline lib/screens/splash/splash.dart]:
```suggestion
    return Scaffold(
      body: Stack(
        children: [
          ScreenBackground(elementId: 0),
          Center(child: ECellLogoAnimation()),
        ],
      ),
    );
```

File: lib/screens/splash/splash.dart [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+          context,
+          MaterialPageRoute(
+              builder: (BuildContext context) => BlocProvider(
+                    create: (_) => LoginCubit(APILoginRepository()),
+                    child: LoginScreen(),
+                  )));
+    } else {
+      Navigator.pushReplacement(
+          context, MaterialPageRoute(builder: (BuildContext context) => HomeScreen()));
+    }
```
Comment (werainkhatri) [inline lib/screens/splash/splash.dart]:
```suggestion
  Navigator.pushReplacement(
    context,
    MaterialPageRoute(
      builder: (_) => (token == null)
          ? BlocProvider(
              create: (_) => LoginCubit(APILoginRepository()),
              child: LoginScreen(),
            )
          : HomeScreen(),
    ),
  );
```

**Review by werainkhatri — CHANGES_REQUESTED** [review]
change the name of ECellAnimation widget file to `ecell_animation.dart`

File: lib/screens/splash/splash.dart (hunk 1 above)
Comment (Hard-Coder05) [reply lib/screens/splash/splash.dart]: done

File: lib/screens/splash/splash.dart (hunk 3 above)
Comment (Hard-Coder05) [reply lib/screens/splash/splash.dart]: done

File: lib/screens/splash/splash.dart (hunk 2 above)
Comment (Hard-Coder05) [reply lib/screens/splash/splash.dart]: done

File: lib/screens/splash/splash.dart (hunk 4 above)
Comment (Hard-Coder05) [reply lib/screens/splash/splash.dart]: done

File: lib/screens/splash/splash.dart (hunk 5 above)
Comment (Hard-Coder05) [reply lib/screens/splash/splash.dart]: done

File: lib/screens/splash/splash.dart (hunk 2 above)
Comment (werainkhatri) [reply lib/screens/splash/splash.dart]: `1.` is not done yet.

File: lib/screens/splash/splash.dart [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+class _SplashScreenState extends State<SplashScreen> {
+  @override
+  void initState() {
+    super.initState();
+    _loadWidget();
+  }
+
+  _loadWidget() async {
+    await init();
+    return Timer(Duration(seconds: D.splashDelay), navigationPage);
```
Comment (werainkhatri) [inline lib/screens/splash/splash.dart]:
```suggestion
    Timer(Duration(seconds: D.splashDelay), navigationPage);
```

**Review by werainkhatri — APPROVED** [review]

**Pull request merged**